server.js

```diff
@@ -1,38 +1,77 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
-
-// Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
-
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
-
-// Start the server
+import express from "express";
+import bodyParser from "body-parser";
+import cors from "cors";
+import mongoose from "mongoose";
+
+import data from "./data/netflix-titles.json";
+
+const listEndpoints = require("express-list-endpoints");
+
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/shows";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+const port = process.env.PORT || 8080;
+const app = express();
+
+app.use(cors());
+app.use(bodyParser.json());
+
+const Show = mongoose.model("Show", {
+  show_id: Number,
+  title: String,
+  director: String,
+  cast: String,
+  country: String,
+  date_added: String,
+  release_year: Number,
+  rating: String,
+  duration: String,
+  listed_in: String,
+  description: String,
+  type: String,
+});
+
+if (process.env.RESET_DATABASE) {
+  const seedDataBase = async () => {
+    await Show.deleteMany({});
+    data.forEach((item) => {
+      const newShow = new Show(item);
+      newShow.save();
+    });
+  };
+  seedDataBase();
+}
+
+app.get("/", (req, res) => {
+  res.send(listEndpoints(app));
+});
+
+app.get("/shows", async (req, res) => {
+  const shows = await Show.find();
+  res.json(shows);
+});
+
+app.get("/shows/:title", async (req, res) => {
```

Comment:
`title` is not a good key for extracting single document, unfortunately. When you use it like this you returning a bad set of data because you can cut off a big portion of it. You can have a lot of shows with the same title, for example, remakes or title is so generic it is reused hundreds of time.

You need to use field which is always unique for every document, usually, it is `id` field.